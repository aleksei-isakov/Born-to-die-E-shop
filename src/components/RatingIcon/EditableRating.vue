<template>
  <div class="rating-group">
    <div v-for="starIndex in starsCount" :key="starIndex" class="stars-wrapper">
      <label
        :aria-label="`${starIndex} stars`"
        class="rating__label"
        :for="`rating-${starIndex}`"
        ><i
          :class="[
            'rating__icon fa fa-star',
            { 'rating__icon--colored': starIndex <= rating }
          ]"
        ></i
      ></label>
      <input
        :id="`rating-${starIndex}`"
        class="rating__input"
        :value="starIndex"
        type="radio"
        name="rating"
        @input="onInputChangeRating(starIndex)"
      />
    </div>
  </div>
</template>

<script>
const STARS_COUNT = 5;

export default {
  name: 'EditableRating',

  props: {
    rating: {
      type: Number,
      default: 0
    }
  },

  data() {
    return {
      starsCount: STARS_COUNT
    };
  },

  methods: {
    onInputChangeRating(starIndex) {
      this.$emit('change-rating', starIndex);
    }
  }
};
</script>

<style lang="scss" scoped>
.rating-group {
  display: inline-flex;
}

.stars-wrapper {
  padding: 0 1px;
}

.rating__icon {
  pointer-events: none;
  color: #ddd;
}

.rating__input {
  position: absolute;
  left: -9999px;
  color: #ddd;

  &--none {
    display: none;
  }
}

.rating__label {
  cursor: pointer;
  font-size: 25px;
}

.rating__icon--colored {
  color: #ffa500;
}

.stars-wrapper .rating__input:checked ~ .stars-wrapper {
  color: #ddd;
}

.rating-group:hover .stars-wrapper .rating__icon {
  color: #ffa500;
}

.stars-wrapper:hover ~ .stars-wrapper .rating__icon {
  color: #ddd;
}
</style>
