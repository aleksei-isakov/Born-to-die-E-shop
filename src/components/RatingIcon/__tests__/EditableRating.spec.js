import { mount } from '@vue/test-utils';
import EditableRating from '@/components/RatingIcon/EditableRating';

let wrapper;

beforeEach(() => {
  wrapper = mount(EditableRating);
});

describe('EditableRating', () => {
  it('is should render rating icon for user', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
