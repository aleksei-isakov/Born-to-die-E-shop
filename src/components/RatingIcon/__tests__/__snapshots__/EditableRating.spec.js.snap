// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableRating is should render rating icon for user 1`] = `
<div class="rating-group">
  <div class="stars-wrapper"><label aria-label="1 stars" for="rating-1" class="rating__label"><i class="rating__icon fa fa-star"></i></label> <input id="rating-1" type="radio" name="rating" class="rating__input" value="1"></div>
  <div class="stars-wrapper"><label aria-label="2 stars" for="rating-2" class="rating__label"><i class="rating__icon fa fa-star"></i></label> <input id="rating-2" type="radio" name="rating" class="rating__input" value="2"></div>
  <div class="stars-wrapper"><label aria-label="3 stars" for="rating-3" class="rating__label"><i class="rating__icon fa fa-star"></i></label> <input id="rating-3" type="radio" name="rating" class="rating__input" value="3"></div>
  <div class="stars-wrapper"><label aria-label="4 stars" for="rating-4" class="rating__label"><i class="rating__icon fa fa-star"></i></label> <input id="rating-4" type="radio" name="rating" class="rating__input" value="4"></div>
  <div class="stars-wrapper"><label aria-label="5 stars" for="rating-5" class="rating__label"><i class="rating__icon fa fa-star"></i></label> <input id="rating-5" type="radio" name="rating" class="rating__input" value="5"></div>
</div>
`;
